subsys/logging/log_backend_rtt.c

```diff
@@ -185,7 +185,7 @@ static int data_out_block_mode(u8_t *data, size_t length, void *ctx)
 
 		if (ret) {
 			on_write(retry_cnt);
-		} else {
+		} else if (host_present) {
```

Comment:
IMHO from time to time host presence should be rechecked.

Reply:
it will be. If `ret!=0` it means that host read the buffer and we were able to put some data to the buffer. In `on_write` `host_present` is set to true.